File: DebugClients/Ruby/DebugClient.rb
# -*- coding: utf-8 -*-

# Copyright (c) 2005 - 2015 Detlev Offenbach <detlev@die-offenbachs.de>
#

=begin edoc
File implementing a debug client.
=end

# insert path to ourself in front of the search path
$:.insert(0, File.dirname($0))

require 'Debuggee'
require 'AsyncIO'
require 'DebugClientBaseModule'

class DebugClient
=begin edoc
Class implementing the client side of the debugger.
=end
    include AsyncIO
    include DebugClientBase
    
    def initialize
=begin edoc
Constructor
=end
        initializeAsyncIO
        initializeDebugClient
        
        @variant = "No Qt-Version"
    end
end

# We are normally called by the debugger to execute directly

if __FILE__ == $0
    debugClient = DebugClient.new()
    debugClient.main()
end

File: DebugClients/Python/DebugClient.py
# -*- coding: utf-8 -*-

# Copyright (c) 2003 - 2015 Detlev Offenbach <detlev@die-offenbachs.de>
#

"""
Module implementing a Qt free version of the debug client.
"""

from AsyncIO import AsyncIO
from DebugBase import DebugBase
import DebugClientBase


class DebugClient(DebugClientBase.DebugClientBase, AsyncIO, DebugBase):
    """
    Class implementing the client side of the debugger.
    
    This variant of the debugger implements the standard debugger client
    by subclassing all relevant base classes.
    """
    def __init__(self):
        """
        Constructor
        """
        AsyncIO.__init__(self)
        
        DebugClientBase.DebugClientBase.__init__(self)
        
        DebugBase.__init__(self, self)
        
        self.variant = 'Standard'

# We are normally called by the debugger to execute directly.

if __name__ == '__main__':
    debugClient = DebugClient()
    debugClient.main()

#
# eflag: FileType = Python2
